addons/website_sale/views/templates.xml

```diff
@@ -713,6 +713,28 @@
     </t>
     </template>
 
+    <template id="alternative_products" name="Alternative Products" inherit_id="website_sale.product" active="True">
+        <div itemprop="description" position="after">
+            <div class="oe_structure oe_structure_solo oe_unremovable oe_unmovable" id="oe_structure_website_sale_recommended_products" t-ignore="true" t-if="product.alternative_product_ids">
+                <section data-snippet="s_dynamic_snippet_products"
```

Comment:
> I looked into the "outdated block" thing for a few hours today.
I still can not reproduce it consistently locally so no way to fix .. However I can confirm that the issue is not directly related to this change.
It may happen to other snippets as well.

@Williambraecky This is because of this `data-snippet` which is wrong: if you drop a "Products" snippet by yourself, you can see that the added data-snippet is `s_dynamic_snippet`, not `s_dynamic_snippet_products`. In your branch you thus have the outdated message when this is added automatically by filling the alternative product field in a form view and you don't have the message when you add the snippet by hand.

Now, I wonder if it is an issue that there is no reference to `s_dynamic_snippet_products` when using that snippet. I'll create a task about that to investigate. Meanwhile, it's more consistent (and maybe entirely right) if you use `s_dynamic_snippet` as data-snippet here.

Reply:
Is it definitely linked to the `data-snippet` however I do not agree with what you are saying.

This is what I get when I drop a dynamic snippet of type products on the shop page for example:
![Capture d’écran 2022-07-18 à 16 54 57](https://user-images.githubusercontent.com/4508773/179539712-1db9d9f5-713e-4fdc-8b7d-631a8466ebd8.png)

The issue seems to be way further where somehow the content for our dynamic content has `data-snippet="s_dynamic_snippet"`

Reply:
If you have a video of that behavior (in the master branch, not yours of course), that would help us :+1: